js/islandora_book_reader.js

```diff
@@ -219,11 +219,34 @@
   IslandoraBookReader.prototype.getPageURI = function(index, reduce, rotate) {
     if (typeof this.settings.pages[index] != 'undefined') {
       var resource_uri = this.settings.pages[index].uri;
+      if (this.getPageFileSize < this.settings.minimumFileSize) {
+        resource_uri = this.settings.pages[index].backupUri;
+      }
       return this.getDjatokaUri(resource_uri);
     }
   }
 
   /**
+   * Gets the size of the image at the given URI.
+   * Using XMLHttpRequest so that we don't have to load the whole image.
+   *
+   * @param string uri
+   *   The URI.
+   *
+   * @return int
+   *   The size of the file.
+   */
+  IslandoraBookReader.prototype.getPageFileSize = function(uri) {
+    var xhr = new XMLHttpRequest();
+    xhr.open('HEAD', uri, true);
```

Comment:
Running this async and then trying to use it right away will have misses.  I would keep it async and use a callback. You should probably move to using jQuery's API; it is what most folks are used to reading.